subsys/net/l2/openthread/openthread.c

```diff
@@ -424,32 +430,35 @@ static int openthread_init(struct net_if *iface)
 
 	__ASSERT(ot_context->instance, "OT instance is NULL");
 
+//	if (!IS_ENABLED(CONFIG_OPENTHREAD_RCP)) { // TODO: check whether required
 	if (IS_ENABLED(CONFIG_OPENTHREAD_SHELL)) {
 		platformShellInit(ot_context->instance);
 	}
 
-	if (IS_ENABLED(CONFIG_OPENTHREAD_NCP)) {
+	if (IS_ENABLED(CONFIG_OPENTHREAD_RCP)) {
 		otNcpInit(ot_context->instance);
```

Comment:
Same remark as concerning Kconfig options' names. Please find some common naming convention to avoid misunderstanding.

Reply:
As discussed will go for `CONFIG_OPENTHREAD_COPROCESSOR`.

Reply:
Ok, but in this place my remark applies to the otNcpInit() method name. I would like to stay consistent and change it to otCoProcessorInit(), but as I guess it would also require contributing to the OpenThread. What do you think about this @rlubos ?

Reply:
Given that `otNcpInit()` is part of the OT API, I don't think there's anything we can do about it in this PR (but I don't find it to be a block either). Of course, it would be nice to have a more generic name, but I'd leave it to upstream community to sort this out (they have the same issue in the `ncp` app, which could be configured either as NCP or RCP).

Reply:
Yeah, looking further, there are a lot of places in OpenThread, where NCP name is used and it also applies to RCP, but I think you are right. It's not good idea to change everything at once and for now it is not a problem at all.